### DISCUSSION TITLE
Please add checks

### DISCUSSION THREAD

**Discussion by ghost** [blocklistproject/Lists#352] [Ideas]
1. Please make sure the domains are registered.

**Comment by iam-py-test** [comment]


**Reply by ghost** [reply]
That's not what I am saying, use a who's is API to see if it's registered,
If the domain itself is available for sale than remove it from the list

**Reply by fishcharlie** [reply]
@Prajwal-Koirala I have never seen a Whois API to determine if the domain itself is available for sale. Domain names normally just change DNS query hosts when they are made available for sale.

Even if we were able to gather the information about determine if a domain name has been sold, there is no guarantee that the website is now used for valid purposes. This would just open the door to scammers selling the domains back and forth to each other to avoid being listed on this blocklist.

I'm open to an idea like this. I think it has potential. But it needs to be flushed out a bit more. Some of these details need to be thought through before we can consider implementing a system like this.

**Reply by fishcharlie** [reply]
@Prajwal-Koirala @iam-py-test I'm not sure what is going on but GitHub isn't showing your comments but I got an email about them. Maybe they were deleted?

@Prajwal-Koirala Anyways, although I appreciate the suggestion, and brainstorming new ideas is great, I think you are oversimplifying this a great deal.

I'm not a lawyer, so this is not legal advice. However, the [Virus Total TOS](https://support.virustotal.com/hc/en-us/articles/115002145529-Terms-of-Service) states: `You agree that you may not use or attempt to... Copy, reproduce, alter, modify, create derivative works, publicly display, republish, upload, post, transmit, resell or distribute in any way material, information or functionalities from the Service`.

From how I read and interpret that, using the API in that way, would completely violate their Terms of Service. That doesn't even take into account @iam-py-test's comments about accuracy. Can you point to examples where domains are not on the correct category list? I can only think of 1 example that I've seen.

In terms of verifying that a domain is registered. I think this is an extreme edge case. How many domains have been added to the list that are not registered? I'm unaware of a single example of this occurring. Possibly domains that were added before have now expired. But I'm unaware of a single instance of a domain being added that isn't even registered. Can you please provide an example of that occurring?

I think it's tough to say `but this is a way better system than what majority of the people are doing now` without providing examples and pointing to instances of problems that you are trying to solve. I'm not sure what problem you are trying to solve.

**Reply by iam-py-test** [reply]
I moved my other comment to here and deleted it:

I saw the email, but as I could not find the comment, I deleted my comment. (I thought the email was just a malfunction)
However I still think auto-domain-rating systems can have false positives.
We could use more than one system, that way we have more accuracy.
A few other possibilities:

  1.  Don’t auto-delete domains that look offline, just move them to somewhere else until they can be verified to be offline
  2.  Have a list of maybe-offline domains and then have the community verify them
  3.  Remove domains if their DNS and WHOIS have been down for more than a certain amount of time
There also may be domains which will never be used for anything else other than malware/phishing/porn (i.e. domains like porn.com) and can just be left in the list incase someone starts it back up.

I think that their API is ok as it is made by VirusTotal and is authorized by them

Edit: For some reason, when I respond via email, it creates a new comment

**Reply by fishcharlie** [reply]
@iam-py-test It doesn't look like VirusTotal is owned by Google anymore. They used to be, but now they are owned by Chronicle. Do you have evidence that this use is authorized by them? Again, I'm not a lawyer, but much better to be cautious in these instances.

**Reply by fishcharlie** [reply]
All of that being said. I think the 2nd point that @iam-py-test made kinda relates to some of the ideas I brought up in #355. I think manually verifying is a good idea.

**Reply by ghost** [reply]
```
2021/06/30 11:29:22 Invalid domain suffix: cookiescript.infos https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
2021/06/30 11:29:20 Invalid domain suffix: analytics.bl https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
2021/06/30 11:29:48 Invalid domain suffix: a4.tl2 https://blocklistproject.github.io/Lists/alt-version/ads-nl.txt
2021/06/30 11:29:49 Invalid domain suffix: nomoreransom.coin https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
2021/06/30 11:29:50 Invalid domain suffix: accounts.pkr.com.invalid https://blocklistproject.github.io/Lists/alt-version/ads-nl.txt
2021/06/30 11:49:55 Invalid domain suffix: cialisn.comparent https://blocklistproject.github.io/Lists/alt-version/drugs-nl.txt
2021/06/30 11:50:22 Invalid domain suffix: adfs.uky.edu.adfs.ls.client.request.id.984b9da9.79f04e2b82b3cec190911989.usernamewawsignin1.0wtrealmurn3afederation.3amicrosofto https://blocklistproject.github.io/Lists/alt-version/phishing-nl.txt
2021/06/30 11:51:14 Invalid domain suffix: anz.com.9ov5c7k7ydvyj1vf8tap0efqh.9ov5c7k7ydvyj1vf8tap0efqh.9ov5c7k7ydvyj1vf8tap0efqh.9ov5c7k7ydvyj1vf8tap0efqh.9ov5c7k7ydvyj1vf https://blocklistproject.github.io/Lists/alt-version/phishing-nl.txt
```
I can provide way more but here is a small sample set.

**Reply by iam-py-test** [reply]
These could be typos:
`cookiescript.infos` could be `cookiescript.info`.
`cookiescript.info` has a lot of detections on [VirusTotal](https://www.virustotal.com/gui/url/9f26c3bb951801b00fe5f37033c2f0523cfa1a37b1eddfd39a77a22f3a167f07/detection) and [this](https://labs.sucuri.net/blacklist/info/?domain=cookiescript.info) says it hosts malware
Also, what tool did you use?

**Reply by iam-py-test** [reply]
I think that maybe this repo should ignore new domains for a while and review every entry (I will gladly help in my spare time)

**Reply by fishcharlie** [reply]
@Prajwal-Koirala @iam-py-test 

I will be locking this discussion if profanity continues. That is not the purpose of this discussion and only distracts from the discussion.

I personally fail to see the issue with domains that are not registered being included in this list. What is the downside here? Why is this such an urgent issue? Sure it increases file size of these lists, but only by a fraction of the total list size. @Prajwal-Koirala has failed to provide proof or evidence that these domains are now clean and used for legit purposes. Just because a domain is no longer valid or never was valid, doesn't necessarily mean it's bad that it was included in the list.

I personally believe the bar for site removal should be having clean content (**NOT** being unregistered).

As I mentioned in other threads. No list is perfect. This project I truly believe is improving everyday and is truly setup for great success in the future. And these lists are **far better** than nothing. That isn't to say these issues shouldn't be resolved. But getting so heated about it doesn't make sense. It's important to put the big picture into perspective. And the big picture is this project has an INCREDIBLE foundation, and is way better than no lists at all. A few unregistered domains or typos is not a huge deal. 100s of issues when we have millions of domains is **NOTHING**. 

> I think that maybe this repo should ignore new domains for a while and review every entry

I do not support this. There is a discussion at #355 about doing an audit and reviewing entries. Please comment there if you have thoughts. However ignoring new domains will not serve any purpose. Especially when new domains are human tested and validated.

> I will gladly help in my spare time

🎉🎉🎉

**Reply by blocklistproject** [reply]
> ```
> 
> 2021/06/30 11:29:22 Invalid domain suffix: cookiescript.infos https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
> 
> 2021/06/30 11:29:20 Invalid domain suffix: analytics.bl https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
> 
> 2021/06/30 11:29:48 Invalid domain suffix: a4.tl2 https://blocklistproject.github.io/Lists/alt-version/ads-nl.txt
> 
> 2021/06/30 11:29:49 Invalid domain suffix: nomoreransom.coin https://blocklistproject.github.io/Lists/alt-version/crypto-nl.txt
> 
> 2021/06/30 11:29:50 Invalid domain suffix: accounts.pkr.com.invalid https://blocklistproject.github.io/Lists/alt-version/ads-nl.txt
> 
> ```
> 
> I can provide way more but heree is a small sample set.

I saw your original post including the profanity. This is your ONLY warning, if you are not able to provide constructive feedback without profanity and/or attacks you will be reported and blocked from this repo. Everyone that works on this project does so on their own free time often taking time away from families and hobbies. For you to get on here and conduct yourself in the way you have is unprofessional and I will not let it slide. Thank you and have a great day.

**Reply by iam-py-test** [reply]
@blocklistproject 
Is this project the same as the _BlockList_ on VirusTotal or is that something else?

**Reply by blocklistproject** [reply]
We have no affiliation with VirusTotal. Can you link to what you are referring too?

**Reply by iam-py-test** [reply]
https://www.virustotal.com/gui/url/9aa242283abf9b5c06c7df629bd4fcecb34ad6fdd04ea0f9365c5ab0eeb7b2a5/detection (or any VirusTotal page) under Bfore.Ai PreCrime

**Reply by blocklistproject** [reply]
I see that. That is not us.

**Reply by iam-py-test** [reply]
I wonder who it is
When I looked it up, I just got your site

**Reply by ghost** [reply]
Need to be better in ML so I can train a bot that looks for this, so list gen is automated.

**Reply by fishcharlie** [reply]
@Prajwal-Koirala Again, I think you are oversimplifying here. Some lists would be straight forward to automate, such as the porn list. Fairly easy to write a script, gather images or videos from the site, and maybe even text, and run it through AWS Rekognition to detect explicit content.

Even at that tho, what happens if the site is temporarily offline? Or down for maintenance? In those events it would detect the site as clean, even tho it's not. Lots of edge cases to consider.

But, when you get into phishing websites, or ads based domains. This becomes a lot more difficult. Even humans sometimes have difficulty determining if a site is a phishing site. Even the most advanced ML models are not as advanced as humans (yet).

I'm not opposed to people exploring this idea. ML is a growing field, and this could totally be possible. This is an open source project and pull requests are always welcome. However just based on the comments you've made so far, I think you are vastly underestimating the challenges in deploying a system to achieve this in an accurate fashion.

**Reply by iam-py-test** [reply]
> @Prajwal-Koirala Again, I think you are oversimplifying here. Some lists would be straight forward to automate, such as the porn list. Fairly easy to write a script, gather images or videos from the site, and maybe even text, and run it through AWS Rekognition to detect explicit content.
> 
> Even at that tho, what happens if the site is temporarily offline? Or down for maintenance? In those events it would detect the site as clean, even tho it's not. Lots of edge cases to consider.
> 
> But, when you get into phishing websites, or ads based domains. This becomes a lot more difficult. Even humans sometimes have difficulty determining if a site is a phishing site. Even the most advanced ML models are not as advanced as humans (yet).
> 
> I'm not opposed to people exploring this idea. ML is a growing field, and this could totally be possible. This is an open source project and pull requests are always welcome. However just based on the comments you've made so far, I think you are vastly underestimating the challenges in deploying a system to achieve this in an accurate fashion.

We could make an _lite_ or _online_ list and have it updated via Python and GitHub actions.
I can copypaste what I use and modify it a bit.
If it is run every time there is a commit + manual running + running every so often, it should catch down times (I hope)

**Reply by fishcharlie** [reply]
@iam-py-test Feel free to submit a pull request and I can review it!

**Reply by iam-py-test** [reply]
@fishcharlie @blocklistproject should the online-only files end with _lite.txt or _online.txt?

**Reply by fishcharlie** [reply]
@iam-py-test Totally up to you at this point. I'll probably have a lot of suggestions. So just make decisions, and put something up and I'll call out things that I see.

**Reply by iam-py-test** [reply]
Which lists should have online-only versions created?

**Reply by fishcharlie** [reply]
@iam-py-test Ok we aren't on the same page. I was under the impression that you were working on an ML based list for any topic as a trial. That is what you quoted in your reply. Each ML based list is going to have MASSIVELY different requirements. The things that'll work for some lists won't work for others. I kinda mentioned that in my comment.

So whatever you want to start with, feel free. But it's not a one size fits all.

Keep in mind, there is a chance this will never be merged in. So if you want to start with a detailed proposal for what you are thinking, that'd work too.

**Reply by iam-py-test** [reply]
> @iam-py-test Ok we aren't on the same page. I was under the impression that you were working on an ML based list for any topic as a trial. That is what you quoted in your reply. Each ML based list is going to have MASSIVELY different requirements. The things that'll work for some lists won't work for others. I kinda mentioned that in my comment.
> 
> So whatever you want to start with, feel free. But it's not a one size fits all.
> 
> Keep in mind, there is a chance this will never be merged in. So if you want to start with a detailed proposal for what you are thinking, that'd work too.

Sorry that I misunderstood what you meant. 
You can look at my work and see if it is valuable or not

**Reply by iam-py-test** [reply]
I created #360.
Does that look ok?

**Comment by blocklistproject** [comment]
I think that @fishcharlie did a good job breaking this down and the complexity that would be involved. There is always ways that we can improve the project and as you can see we are actively working to improve the project.

**Comment by iam-py-test** [comment]
TODO: After https://github.com/blocklistproject/Lists/pull/365 is done, add checks for redirect to known domain-parking-services
